daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -1806,68 +1881,101 @@ object VerifyTestCase {
    * @param expectedDiags                             Expected diagnostics from test cases
    * @param implString                                Implementation string
    */
-  def verifyAllDiagnosticsFound(
+  def verifyDiagnosticsFound(
     actualDiags: Seq[Diagnostic],
-    expectedDiags: Option[ErrorWarningBase],
+    expectedDiags: Option[Seq[ErrorWarningBase]],
+    ignoreUnexpectedWarnings: Boolean,
```

Comment:
Can this parameter be changed to a more generic `ignoreUnexpectedDiags: Boolean`? For example, when the `expectedDiags` are `ExpectedErrors` we would always pass in false (we never ignore unexpected errors). When `expectedDiags` are `ExpectedWarnings` we would pass in the value of `ignoreUnexpectedWarnings`. That way this function doesn't need to have have specific logic for warnings (or validationErrors when we add ignoreUnexpectedValidationErrors). It just has something like:

```scala
val hasUnexpectedDiags = actualDiagsFiltered.isNonEmpty && expectedDiags.isEmpty
if (hasUnexpectedDiags && !ignoreUnexpectedDiags)
  throw new TDMLExecption(found unexpected diags)
```

Reply:
actualDiagsFiltered will be empty if expectedDiags.isEmpty...since it uses the type of the expectDiags for filtering, but I think we can just change that to actualDiags and the logic works.

Unless we want actualDiagsFiltered to return unfiltered actualDiags when we have no expected diags, thoughts on that?

Reply:
If we use actualDiags then I think it will error if we have any diagnostics, not just unexpected diagnostics of the right type. Feels like we should pass in an an enum that says which diag we are checking, or pass in a filter function that is passed in that is used to pick wich diagnostics to check. Then we don't have to rely on the expected diags to know which actual diags to compare against.

Reply:
Hmm you're right...ok, will add enum called diagnosticType

Reply:
Another alternative, if you update everything to calls checkDiagnostisMesssages you could do the filter in there, since each call to verifyDiagnosticFound knows which type it's verifying.